src/main/scala/nl.knaw.dans.easy.bagstore/ItemId.scala

```diff
@@ -33,12 +33,26 @@ abstract class ItemId(val uuid: UUID) {
 }
 
 object ItemId {
+  // accept both upper and lower case? https://stackoverflow.com/questions/8258480/type-of-character-generated-by-uuid
+  val uuidRegex = "[a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{12}"
+
   def fromString(s: String): Try[ItemId] = Try {
     s.split("/", 2) match {
-      case Array(uuidStr) => BagId(UUID.fromString(uuidStr))
+      //FIXME is this the best spot to validate uuid?
+      case Array(uuidStr) => BagId(UUID.fromString(validateUuid(uuidStr)))
       case Array(uuidStr, path) =>
-        FileId(UUID.fromString(uuidStr), Paths.get(URLDecoder.decode(path, "UTF-8")))
+        FileId(UUID.fromString(validateUuid(uuidStr)), Paths.get(URLDecoder.decode(path, "UTF-8")))
```

Comment:
```suggestion
        FileId(UUID.fromString(validateUuid(uuidStr.toZLowerCase)), Paths.get(URLDecoder.decode(path, "UTF-8")))
```

This makes the regular expression easier.

Reply:
Agreed